pfcpiface/conn.go

```diff
@@ -74,26 +77,29 @@ func (pConn *PFCPConn) startHeartBeatMonitor() {
 	hbCtx, hbCancel := context.WithCancel(pConn.ctx)
 	pConn.hbCtxCancel = hbCancel
 
-	log.Infoln("Starting Heartbeat timer")
+	hbLog := log.WithFields(log.Fields{
+		"interval": pConn.upf.hbInterval,
+	})
+	hbLog.Infoln("Starting Heartbeat timer")
 
-	heartBeatTimer := time.NewTimer(pConn.upf.hbInterval)
+	heartBeatExpiryTimer := time.NewTicker(pConn.upf.hbInterval)
 
 	for {
 		select {
 		case <-hbCtx.Done():
 			log.Infoln("Cancel HeartBeat Timer", pConn.RemoteAddr().String())
-			return
+			heartBeatExpiryTimer.Stop()
 
-		case <-heartBeatTimer.C:
+			return
+		case <-pConn.hbReset:
+			heartBeatExpiryTimer.Reset(pConn.upf.hbInterval)
+		case <-heartBeatExpiryTimer.C:
 			log.Traceln("HeartBeat Interval Timer Expired", pConn.RemoteAddr().String())
 
 			r := pConn.getHeartBeatRequest()
 
-			reply, timeout := pConn.sendPFCPRequestMessage(r)
-
-			if reply != nil {
-				heartBeatTimer = time.NewTimer(pConn.upf.hbInterval)
-			} else if timeout {
+			if _, timeout := pConn.sendPFCPRequestMessage(r); timeout {
+				heartBeatExpiryTimer.Stop()
 				pConn.Shutdown()
 			}
```

Comment:
Should the ticker be reset on success? We send a hb request and got a reply, so the connection is live?

Reply:
The ticker does not have to be reset, but continues sending new signals until stopped.